docs/reference/migration/migrate_7_0/migrate_to_java_time.asciidoc

```diff
@@ -0,0 +1,309 @@
+[[migrate_to_java_time]]
+=== Java Time Migration Guide
+
+Since version 7.0 Elasticsearch is using java.time as an underlying implementation of dates parsing, formatting and calculations.
+java.time is in many aspects very similar to joda-time which we previously used. However there are some differences and this guide is meant to
+help you to become compatible.
+
+==== Identifying if you are affected
+You are only affected if you have used any custom date formats in your mappings, pipelines or templates.
+Most of the custom date formats are compatible, but with some there is a change of meaning or missing. More on that below.
+A custom format is specified on a date field like this:
+[source,console]
+--------------------------------------------------
+PUT my_index
+{
+  "mappings": {
+    "properties": {
+      "date": {
+        "type":   "date",
+        "format": "YYYY/MM/dd HH:mm:ss||YYYY/MM/dd||epoch_millis"
+      }
+    }
+  }
+}
+--------------------------------------------------
+
+To make sure if any of the date formats is deprecated run deprecation API or kibana upgrade assistant.
```

Comment:
To see if your date format is impacted, use the <<migration-api-deprecation,deprecation info API>>
or the {kibana-ref}/upgrade-assistant.html[Kibana upgrade assistant]